src/flask_container_scaffold/app_scaffold.py

```diff
@@ -2,76 +2,99 @@
                                    lstrip_blocks=True)
-        self.app = app or Flask(name, instance_relative_config=True)
+        self.app = (app or
+                    Flask(name,
+                          instance_relative_config=instance_relative_config,
+                          instance_path=instance_path))
         self.config = config
+        self.silent = not settings_required
+        self.relative = instance_relative_config
         self._init_app()
 
     def _init_app(self):
-        self._setup_instance_folder()
-        self._configure_app()
+        self._load_flask_settings()
+        self._load_custom_settings()
 
-    # NOTE: This method is old, and likely to go away, unless
-    # we make it for dev mode only, as a production app should not
-    # need to create folders.
-    def _setup_instance_folder(self):
-        # Ensure the instance folder exists
-        try:
-            os.makedirs(self.app.instance_path)
-        except OSError:
-            pass
 
-    def _configure_app(self):
-        if self.config is None:
-            # load the instance config, if it exists
-            self.app.config.from_pyfile('settings.cfg', silent=True)
-        else:
+    def _load_flask_settings(self):
+        """
+        This loads the 'core' settings, ie, anything you could set directly
+        on a Flask app. These can be specified in the following order, each
+        overriding the last, if specified:
+        - via config mapping
+        - via Flask settings.cfg file
+        - via environment variable 'FLASK_SETTINGS'
+        """
+        config_not_loaded = True
+        if self.config is not None:
             # load the config if passed in
             self.app.config.from_mapping(self.config)
-            # Next, load from override file, if specified
-        if os.environ.get('SETTINGS'):
-            self.app.config.from_envvar('SETTINGS')
-        # Next, load env vars directly, which will override
-        # any previous settings.
-        self.setup_core_config_from_env()
-        # Loop again, making sure everything in env_list is set somehow
-        # on the config object.
-        self.setup_core_config_from_env(verify=True)
-        # Load the defaults yaml file
-        default_conf = (self.app.config.get('DEFAULTS_FILE') or False)
-        if default_conf:
-            self.app.config.update(
-                {'default_params': load_yaml(default_conf)})
+            config_not_loaded = False
+        # load the instance config, if it exists and/or is required
+        try:
+            self.app.config.from_pyfile('settings.cfg', silent=self.silent)
+            config_not_loaded = False
+        except:
+            config_not_loaded = True
+            pass
+        # Do any additional config specified in the FLASK_SETTINGS file,
+        # if it exists. We only want to fail in the case where settings are
+        # required by the app.
+        if ((config_not_loaded and self.silent == False)
```

Comment:
Shouldn't that be silent == True, if I understand the comment correctly? I'm not sure how the fail condition happens or what failing means here. Shouldn't we load FLASK_SETTINGS anyway and check after for config_not_loaded? Maybe I am misunderstanding the use of silent. Is it to force the use of a settings.cfg file?